### PR TITLE
MueLu: Add simple unit test for SegregatedAFactory and modernize code

### PR DESCRIPTION
**Pull request by maxfirmbach**
@trilinos/muelu

## Motivation

Adding unit testing for the `SegregatedAFactory` and updating the code a little bit. Still or/and again in use for contact/interface problems to avoid aggregation over a certain interface.

I didn't update any loops or data-structures to `Kokkos` as I'm not so familiar with it.

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to #9743
* Part of 
* Composed of 

## Stakeholder Feedback

Monolithic multigrid for contact/interface problems.

## Additional Information

Might be interesting for @mayrmt @csteimer

### CODE REVIEW

**Comment by cgcgcg** [comment]
Instead of reading the matrix in from file, you could create it on-the-fly since it's pretty small.

**Comment by maxfirmbach** [comment]
@cgcgcg I updated the unit-test, matrix is now created at runtime without reading in an additional file.

File: packages/muelu/test/unit_tests/SegregatedAFactory.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      graph->insertGlobalIndices(1, Teuchos::tuple<GlobalOrdinal>(0, 1, 2, 3));
+      graph->insertGlobalIndices(2, Teuchos::tuple<GlobalOrdinal>(0, 1, 2, 3, 4, 5));
+      graph->insertGlobalIndices(3, Teuchos::tuple<GlobalOrdinal>(0, 1, 2, 3, 4, 5));
+      graph->insertGlobalIndices(4, Teuchos::tuple<GlobalOrdinal>(2, 3, 4, 5, 6));
+      graph->insertGlobalIndices(5, Teuchos::tuple<GlobalOrdinal>(2, 3, 4, 5, 6));
+      graph->insertGlobalIndices(6, Teuchos::tuple<GlobalOrdinal>(4, 5, 6));
+      graph->fillComplete();
+
+      RCP<Matrix> A = MatrixFactory::Build(graph.getConst());
+      A->setAllToScalar(1.0);
```
Comment (mayrmt) [inline packages/muelu/test/unit_tests/SegregatedAFactory.cpp:103]:
```suggestion
      A->setAllToScalar(TST::one());
```

File: packages/muelu/test/unit_tests/SegregatedAFactory.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
```
Comment (mayrmt) [inline packages/muelu/test/unit_tests/SegregatedAFactory.cpp:40]:
```suggestion
```

**Review by mayrmt — DISMISSED** [review]

**Review by mayrmt — APPROVED** [review]

**Comment by GrahamBenHarper** [comment]
@maxfirmbach it looks like your code is fine, but the randomly failing ROL test caused your PR to fail. I put a retest on so that it'll hopefully pass this time :)

**Comment by GrahamBenHarper** [comment]
Marking the relevant issue for posterity: #11621

**Comment by mayrmt** [comment]
@maxfirmbach Please let me know, if this is ready to be merged from your point of view.

**Review by cgcgcg — APPROVED** [review]
LGTM

**Comment by maxfirmbach** [comment]
Feel free to merge this :+1:.

**Pull request merged**